salt/syspaths.py

```diff
@@ -90,11 +92,44 @@
 CLOUD_DIR = os.path.join(INSTALL_DIR, "cloud")
 BOOTSTRAP = os.path.join(CLOUD_DIR, "deploy", "bootstrap-salt.sh")
 
+
+def _get_windows_root_dir():
+    # Try to get the root directory location from the registry
+    # This key will be created by the NullSoft installer
+    # If salt is currently installed in C:\salt and the user performs an
+    # upgrade, then this key will be set to C:\salt
+    # TODO: Probably need to lock down this key in salt.utils.verify.py
+    # TODO: and in the NullSoft installer code
+    root_dir = salt.utils.win_reg.read_value(
+        hive="HKLM", key="SOFTWARE\\Salt Project\\salt", vname="root_dir"
+    )
+    if root_dir["success"]:
+        # Make sure vdata contains something
+        if root_dir["vdata"]:
+            return root_dir["vdata"]
+
+    # If this key does not exist, then salt was not installed using the
+    # new method installer. Could be pip or setup.py or an older version of the
+    # installer.
+    log.debug("Failed to get ROOT_DIR from registry. {}".format(root_dir["comment"]))
+    # Check for C:\salt\conf
+    old_root = "\\".join([os.environ["SystemDrive"], "salt", "conf"])
```

Comment:
Should we be doing any path verification here?

Reply:
I don't think so